gradle/ide.gradle

```diff
@@ -0,0 +1,141 @@
+import org.jetbrains.gradle.ext.Remote
+import org.jetbrains.gradle.ext.JUnit
+
+buildscript {
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+  dependencies {
+    classpath "gradle.plugin.org.jetbrains.gradle.plugin.idea-ext:gradle-idea-ext:0.7"
+  }
+}
+
+apply plugin: org.jetbrains.gradle.ext.IdeaExtPlugin
+
+allprojects {
+  apply plugin: 'idea'
+
+  tasks.named('idea') {
+    doFirst { throw new GradleException("Use of the 'idea' task has been deprecated. For details on importing into IntelliJ see CONTRIBUTING.md.") }
+  }
+}
+
+tasks.register('configureIdeaGradleJvm') {
+  group = 'ide'
+  description = 'Configures the appropriate JVM for Gradle'
+
+  doLast {
+    modifyXml('.idea/gradle.xml') { xml ->
```

Comment:
Why create these files programatically, but not the two that are committed/shared (excluded in gitignore)?

Reply:
We aren't _creating_ these files, we are _editing_ them. These are created by IntelliJ by the import and contain other config, we just want to "tweak" them. The files in `.idea` that I've added to source control are specifically designed to be shared via VCS.

Reply:
Can I suggest a healthy dose of commenting to make this clear to everyone?

Reply:
I've added a javadoc to this method.